### PR TITLE
Implement system metrics collector

### PR DESCRIPTION
**Pull request by girikuncoro**
This collector will collect metrics associated with
NSX-T system health which highly impacted by
controller and manager cluster. In modern NSX-T system, the controller
and manager are included as single appliance cluster. This will
collect 5 metrics: cluster status, cluster node total, cluster node up,
cluster node status, and appliance service.

Signed-off-by: Giri Kuncoro <girikuncoro@gmail.com>

### CODE REVIEW

**Comment by girikuncoro** [comment]
Address #4

File: client/nsxt_client.go [hunk 1]
```diff
@@ -42,3 +43,18 @@ func (c *nsxtClient) GetDhcpStatus(dhcpID string, localVarOptionals map[string]i
 	dhcpServerStatus, _, err := c.apiClient.ServicesApi.GetDhcpStatus(c.apiClient.Context, dhcpID)
 	return dhcpServerStatus, err
 }
+
+func (c *nsxtClient) ReadClusterStatus() (administration.ClusterStatus, error) {
+	clusterStatus, _, err := c.apiClient.NsxComponentAdministrationApi.ReadClusterStatus(c.apiClient.Context, nil)
```
Comment (walbertus) [inline client/nsxt_client.go:58]: for nil `localVarOptionals` are we going to hide it from the interface signature?

File: collector/system_collector.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+func newSystemCollector(apiClient *nsxt.APIClient, logger log.Logger) prometheus.Collector {
+	nsxtClient := client.NewNSXTClient(apiClient, logger)
+	clusterStatus := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "cluster", "status"),
+		"Status of NSX-T system controller and management cluster.",
+		nil,
+		nil,
+	)
+	clusterNodeTotal := prometheus.NewDesc(
```
Comment (walbertus) [inline collector/system_collector.go]: Do we still need total and number of Up nodes?
We already have individual UP/DOWN node, right?

File: collector/system_collector.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	clusterNodeUp := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "cluster_node", "up"),
+		"Total number of NSX-T system cluster nodes currently up.",
+		nil,
+		nil,
+	)
+	clusterNodeStatus := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "cluster_node", "status"),
+		"Status of NSX-T system cluster nodes UP/DOWN.",
+		[]string{"name", "type"},
```
Comment (walbertus) [inline collector/system_collector.go]: If there's no property "name" it's okay to put ip_address as label

File: collector/system_collector.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+		return
+	}
+
+	for _, c := range clusterNodesStatus.ControllerCluster {
+		// No node name property provided, use ip address instead
+		name := c.RoleConfig.ControlPlaneListenAddr.IpAddress
+
+		if strings.ToUpper(c.NodeStatus.ControlClusterStatus.ControlClusterStatus) == "CONNECTED" &&
+			strings.ToUpper(c.NodeStatus.ControlClusterStatus.MgmtConnectionStatus.ConnectivityStatus) == "CONNECTED" {
+			ch <- prometheus.MustNewConstMetric(sc.clusterNodeStatus, prometheus.GaugeValue, 1.0, name, "controller")
```
Comment (walbertus) [inline collector/system_collector.go]: For constant "controller" and "management" should we create two different desc and use constant labels instead? It's in the fourth parameter of NewDesc

**Review by walbertus — COMMENTED** [review]
There's some comment here

File: collector/system_collector.go (hunk 4 above)
Comment (girikuncoro) [reply collector/system_collector.go]: good idea, let me do that

File: client/nsxt_client.go (hunk 1 above)
Comment (girikuncoro) [reply client/nsxt_client.go:58]: let's use YAGNI principle for this https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it
we can throw things that don't matter for this (in this case: localVarOptionals) for ReadClusterStatus method

File: collector/system_collector.go (hunk 2 above)
Comment (girikuncoro) [reply collector/system_collector.go]: good catch, let me remove this, this is redundant

File: client/nsxt_client.go (hunk 1 above)
Comment (walbertus) [reply client/nsxt_client.go:58]: ack, will update existing PR

**Review by walbertus — APPROVED** [review]
lgtm

**Pull request merged**